snippets/Google.Storage.V1.Snippets/StorageClientSnippets.cs

```diff
@@ -12,33 +12,212 @@
+            var destination = "greetings/hello.txt";
+            var contentType = "text/plain";
+
+            using (var stream = File.OpenRead(source))
+            {
+                // IUploadProgress defined in Google.Apis.Upload namespace
+                var progress = new Progress<IUploadProgress>(
+                  p => Console.WriteLine($"bytes: {p.BytesSent}, status: {p.Status}")
+                );
+
+                // ObjectsResource defined in Google.Apis.Storage.v1 namespace
+                // var acl = ObjectsResource.InsertMediaUpload.PredefinedAclEnum.PublicRead // public
+                var acl = ObjectsResource.InsertMediaUpload.PredefinedAclEnum.AuthenticatedRead; // private
+                var options = new UploadObjectOptions { PredefinedAcl = acl };
+                var obj = client.UploadObject(bucketName, destination, contentType, stream, options, progress);
+            }
+            // </UploadObject>
+
+            // want to show the source and destinations in the snippet, but also
+            // want to make sure it matches the one in the fixture
+            Assert.Equal(local, source);
+            Assert.Equal(destination, fixture.HelloWorldName);
+
+            // No need to delete the storage object, as it'll be taken down by the fixture cleanup
+            File.Delete(local);
+        }
+
+        [Fact]
+        void GetObject()
+        {
+            var bucketName = fixture.BucketName;
+
+            // <GetObject>
+            var client = StorageClient.Create();
+            var name = "greetings/hello.txt";
+
+            var obj = client.GetObject(bucketName, name);
+            Console.WriteLine($"Name: {obj.Name}");
+            Console.WriteLine($"Size: {obj.Size}");
+            Console.WriteLine($"ContentType: {obj.ContentType}");
+            Console.WriteLine($"TimeCreated: {obj.TimeCreated}");
+            // </GetObject>
+        }
+
+        [Fact]
+        void GetBucket()
+        {
+            var bucketName = fixture.BucketName;
+
+            // <Buckets.Get>
+            var client = StorageClient.Create();
+
+            var bucket = client.Service.Buckets.Get(bucketName).Execute();
+            Console.WriteLine($"Name: {bucket.Name}");
+            Console.WriteLine($"TimeCreated: {bucket.TimeCreated}");
+            // </Bucket.Get>
+        }
+
+        [Fact]
+        void DeleteObject()
+        {
+            var bucketName = fixture.BucketName;
+
+            // <DeleteObject>
+            var client = StorageClient.Create();
+            var objectName = "greetings/hello.txt";
+
+            client.DeleteObject(bucketName, objectName);
+            // </DeleteObject>
+
+            // want to show the name in the snippet, but also
+            // want to make sure it matches the ones in the fixture
+            Assert.Equal(fixture.HelloWorldName, objectName);
+
+            // need to put the object back as it's used elsewhere
```

Comment:
I think it would be cleaner for this test to create a different file to start with (before the snippet part) and then delete it.